src/App.js

```diff
@@ -1,45 +1,91 @@
 import './styles/reset.css'
 import './styles/index.css'
+import { useState } from 'react'
+import Cart from './components/cart'
+import Store from './components/store'
 
-import initialStoreItems from './store-items'
-
-/*
-Here's what a store item should look like
-{
-  id: '001-beetroot',
-  name: 'beetroot',
-  price: 0.35
-}
-
-What should a cart item look like? 🤔
-*/
+export default function App () {
+  // Setup state here...
 
-console.log(initialStoreItems)
+  const [cart, setCart] = useState([])
+  const [formData, setForm] = useState({
+    filterType: '',
+    sortBy: ''
+  })
 
-export default function App() {
-  // Setup state here...
+  function handleChange (e) {
+    const { name, value } = e.target
+    setForm(x => { return { ...x, [name]: value } })
+  }
+  function addToCart (item) {
+    checkQuantity(item.id) ? IncQuantity(item) : setCart([...cart, { ...item }])
+  }
+  function removeFromCart (item) {
+    item.quantity === 1 ? setCart(x => x.filter(c => c.id !== item.id)) : DecQuantity(item)
+  }
+  function DecQuantity (item) {
+    setCart(x => x.map(c => c.id === item.id ? { ...c, quantity: c.quantity - 1 } : c))
+  }
+  function IncQuantity (item) {
+    setCart(x => x.map(c => c.id === item.id ? { ...c, quantity: c.quantity + 1 } : c))
+  }
+  function checkQuantity (id) {
+    return cart.some(x => x.id === id)
+  }
+  const total = cart.reduce((acc, curr) => acc + (curr.price * curr.quantity), 0)
 
   return (
     <>
       <header id="store">
         <h1>Greengrocers</h1>
         <ul className="item-list store--item-list">
-          {/* Wrtite some code here... */}
+          <Store
```

Comment:
I like that you have done this as a form and have reduced both your filter values to a single state. NIIIIIIIIICE